src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs

```diff
@@ -157,15 +157,46 @@ public ShardedCollectionPreviewOperation(ShardedDatabaseRequestHandler handler,
 
         public HttpRequest HttpRequest => _handler.HttpContext.Request;
 
-        public RavenCommand<StreamResult> CreateCommandForShard(int shardNumber) =>
-            new ShardedCollectionPreviewCommand(_handler, _token.Pages[shardNumber].Start, _token.PageSize);
+        public RavenCommand<StreamResult> CreateCommandForShard(int shardNumber)
+        {
+            var collection = GetCollection();
```

Comment:
I think you already have this in the abstract, no? called `Collection`

Reply:
Thx!